example/tour_guide/ai_assistants.py

```diff
@@ -34,22 +34,17 @@ class TourGuideAIAssistant(AIAssistant):
     name = "Tour Guide Assistant"
     instructions = (
         "You are a tour guide assistant that offers information about nearby attractions. "
-        "The application will capture the user coordinates, and should provide a list of nearby attractions. "
-        "Use the available tools to suggest nearby attractions to the user. "
-        "You don't need to include all the found items, only include attractions that are relevant for a tourist. "
-        "Select the top 10 best attractions for a tourist, if there are less then 10 relevant items only return these. "
-        "Order items by the most relevant to the least relevant. "
-        "If there are no relevant attractions nearby, just keep the list empty. "
-        "Your response will be integrated with a frontend web application therefore it's critical that "
-        "it only contains a valid JSON. DON'T include '```json' in your response. "
+        "You will receive the user coordinates and should use available tools to find nearby attractions. "
+        "Only call the find_nearby_attractions tool once. "
+        "Your response should only contain valid JSON data. DON'T include '```json' in your response. "
         "The JSON should be formatted according to the following structure: \n"
         f"\n\n{_tour_guide_example_json()}\n\n\n"
         "In the 'attraction_name' field provide the name of the attraction in english. "
         "In the 'attraction_description' field generate an overview about the attraction with the most important information, "
         "curiosities and interesting facts. "
         "Only include a value for the 'attraction_url' field if you find a real value in the provided data otherwise keep it empty. "
     )
-    model = "gpt-4o"
+    model = "gpt-4o-2024-08-06"
```

Comment:
Perhaps its best to not hardcode the date here?

Reply:
In this case it makes sense as structured outputs was trained in this version

1 https://openai.com/index/introducing-structured-outputs-in-the-api/
```
A new option for the response_format parameter: developers can now supply a JSON Schema via json_schema, a new option for the response_format parameter. This is useful when the model is not calling a tool, but rather, responding to the user in a structured way. This feature works with our newest GPT-4o models: gpt-4o-2024-08-06, released today, and gpt-4o-mini-2024-07-18. When a response_format is supplied with strict: true, model outputs will match the supplied schema.
```

https://platform.openai.com/docs/models/gpt-4o
```
Latest snapshot that supports [Structured Outputs](https://platform.openai.com/docs/guides/structured-outputs)
```

Reply:
sounds good